openedx/adg/lms/applications/tests/test_forms.py

```diff
@@ -180,58 +162,36 @@ def test_user_application_cover_letter_form_business_line(business_line, button_
 
     data_dict = {
         'business_line': business_line,
-        'button_click': button_click
+        'submit_or_back_clicked': button_click,
+        'interest_in_business': interest_in_business,
     }
-    form = UserApplicationCoverLetterForm(data=data_dict)
 
+    form = BusinessLineInterestForm(data=data_dict)
     assert form.is_valid() == expected
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize(
-    'cover_letter_text, cover_letter_file, expected',
-    [
-        ('Test cover letter text', None, True),
-        (None, 'Test Cover Letter File', True),
-        ('Test cover letter text', 'Test Cover Letter File', False)
-    ],
-    ids=['cover_letter', 'cover_letter_file', 'cover_letter_and_cover_letter_file'])
-def test_user_application_cover_letter_duplication(
-    cover_letter_text, cover_letter_file, expected, mock_cover_letter_file
+@pytest.mark.parametrize('button_click, interest_in_business, expected', [
+    ('back', '', True),
+    ('back', TEST_INTEREST_IN_BUSINESS, True),
+    ('back', TEST_INTEREST_IN_BUSINESS * 300, False),
```

Comment:
Isn't this an overkill just to prepare data for this test case? I mean we are multiplying the string 'Test Interest in Business' to 300. An unnecessary operation, which will result in a string wayyy longer than the max limit

Reply:
In an ideal situation, we will be testing the boundry cases of word limit, i.e. MAX_LIMIT, and MAX_LIMIT + 1

Reply:
It is checking for the word limit.

Reply:
Oh my bad, I thought we were testing the character length. In that case, to make it generic, we can multiply it with the MAX_WORD_LIMIT instead of 300.

Also, even if we are testing words, we should test boundary cases, like 200 words and 201 words

Reply:
@MunamMubashir-arbisoft here as well, we should do the boundary thingy we did for the other validation test

Reply:
We have already tested the validation at the validation method point. We do not need to test it everywhere. Mostly, it scenarios like this, we mock the inner function call to return the valid input. Here, I'm only testing from the form point of view.